crates/core/src/pattern_compiler/compiler.rs

```diff
@@ -567,6 +567,121 @@ fn find_definition_if_exists(
     Ok(None)
 }
 
+pub fn get_dependents_of_target_patterns_by_traversal_from_src(
+    libs: &BTreeMap<String, String>,
+    src: &str,
+    parser: &mut MarzanoGritParser,
+    will_autowrap: bool,
+    target_patterns: &[String],
+) -> Result<Vec<String>> {
+    let mut dependents = <Vec<String>>::new();
+    let node_like = "nodeLike";
+    let predicate_call = "predicateCall";
+
+    let tree = parser.parse_file(src, Some(Path::new(DEFAULT_FILE_NAME)))?;
+
+    let DefsToFilenames {
+        patterns: pattern_file,
+        predicates: predicate_file,
+        functions: function_file,
+        foreign_functions: foreign_file,
+    } = defs_to_filenames(libs, parser, tree.root_node())?;
+
+    let mut traversed_stack = <Vec<String>>::new();
+
+    // gross but necessary due to running these patterns before and after each file
+    let mut stack: Vec<Tree> = if will_autowrap {
```

Comment:
Just leave this logic out - we don't need to worry about autowrap / before patterns for `--watch`.